components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -573,47 +561,14 @@ std::string BraveReferralsService::BuildReferralFinalizationCheckPayload()
   return result;
 }
 
-void BraveReferralsService::FetchReferralHeaders() {
-  net::NetworkTrafficAnnotationTag traffic_annotation =
-      net::DefineNetworkTrafficAnnotation(
-        "brave_referral_headers_fetcher", R"(
-        semantics {
-          sender:
-            "Brave Referrals Service"
-          description:
-            "Fetches referral headers from Brave."
-          trigger:
-            "An update timer indicates that it's time to fetch referral headers."
-          data: "Brave referral headers."
-          destination: WEBSITE
-        }
-        policy {
-          cookies_allowed: NO
-          setting:
-            "This feature cannot be disabled by settings."
-          policy_exception_justification:
-            "Not implemented."
-        })");
-  auto resource_request = std::make_unique<network::ResourceRequest>();
-  resource_request->url =
-      GURL(BuildReferralEndpoint(kBraveReferralsHeadersPath));
-  resource_request->credentials_mode = network::mojom::CredentialsMode::kOmit;
-  resource_request->load_flags = net::LOAD_DO_NOT_SAVE_COOKIES |
-                                 net::LOAD_BYPASS_CACHE |
-                                 net::LOAD_DISABLE_CACHE;
-  network::mojom::URLLoaderFactory* loader_factory =
-      g_browser_process->system_network_context_manager()
-          ->GetURLLoaderFactory();
-  referral_headers_loader_ = network::SimpleURLLoader::Create(
-      std::move(resource_request), traffic_annotation);
-  referral_headers_loader_->SetAllowHttpErrorResults(true);
-  referral_headers_loader_->SetRetryOptions(
-      1, network::SimpleURLLoader::RetryMode::RETRY_ON_NETWORK_CHANGE);
-  referral_headers_loader_->DownloadToString(
-      loader_factory,
-      base::BindOnce(&BraveReferralsService::OnReferralHeadersLoadComplete,
-                     base::Unretained(this)),
-      kMaxReferralServerResponseSizeBytes);
+void BraveReferralsService::SetReferralHeaders() {
+  base::Value headers(base::Value::Type::LIST);
+  headers.Append(CreateReferralHeader(kPartnerEaffName, kPartnerEaffDomains));
+  headers.Append(
+      CreateReferralHeader(kPartnerUpholdName, kPartnerUpholdDomains));
+  headers.Append(
+      CreateReferralHeader(kPartnerGrammarlyName, kPartnerGrammarlyDomains));
+  pref_service_->Set(kReferralHeaders, headers);
```

Comment:
can't we just hardcode the whole value as a string and then parse the `base::Value` from json in one line?

Reply:
this is a style thing, and i prefer what i have. parsing JSON not covered by tests. at least with this we get compile-time type checking

Reply:
so we can remove fake header data from the test and do real checks for real hardcoded domains. This will increase safety and reduce boilerplate.